scripts/create-merged-schema.ts

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env ts-node
+/*
+ * Create a complete JSON schema for tsconfig.json
+ * by merging the schemastore schema with our ts-node additions.
+ * This merged schema can be submitted in a pull request to
+ * SchemaStore.
+ */
+
+import axios from 'axios';
+import * as Path from 'path';
+import * as fs from 'fs';
+
+async function main() {
+  /** schemastore definition */
+  const schemastoreSchema = (await axios.get(
+    'https://schemastore.azurewebsites.net/schemas/json/tsconfig.json',
+    { responseType: "json" }
+  )).data;
+
+  /** ts-node schema auto-generated from ts-node source code */
+  const typescriptNodeSchema = require('../tsconfig.schema.json');
+
+  /** Patch ts-node stuff into the schemastore definition. */
+  const mergedSchema = {
+    ...schemastoreSchema,
+    definitions: {
+      ...schemastoreSchema.definitions,
+      tsNodeDefinition: {
+        properties: {
+          'ts-node': {
+            ...typescriptNodeSchema.definitions.TsConfigOptions,
+            description: typescriptNodeSchema.definitions.TsConfigSchema.properties['ts-node'].description,
+            properties: {
+              ...typescriptNodeSchema.definitions.TsConfigOptions.properties,
+              compilerOptions: {
+                ...typescriptNodeSchema.definitions.TsConfigOptions.properties.compilerOptions,
+                allOf: [{
+                  $ref: '#/definitions/compilerOptionsDefinition/properties/compilerOptions'
+                }]
+              }
+            }
+          }
+        }
+      },
+    },
+    allOf: [
+      ...schemastoreSchema.allOf.slice(0, 4),
```

Comment:
What does this mean? Why 4?

Reply:
Just so it gets spliced into the array at a point that looks nice.  https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/tsconfig.json#L630-L631